pkg/resources/storage_integration.go

```diff
@@ -154,28 +154,33 @@ func CreateStorageIntegration(d *schema.ResourceData, meta any) error {
 		req.WithStorageBlockedLocations(storageBlockedLocations)
 	}
 
-	storageProvider := d.Get("storage_provider").(string)
+	storageProvider := strings.ToUpper(d.Get("storage_provider").(string))
```

Comment:
Nit: We could introduce a `StorageProvider` enum in SDK, but I guess this will be done during the rework.

Reply:
Yeah I didn't want to change much here